### DISCUSSION TITLE
Передача файлов из временной таблицы в простую

### DISCUSSION THREAD

**Discussion by erpic-ru** [totumonline/totum-ru-issues-and-discussions#276] [Как это сделать?] [answered]
Подскажите, пожалуйста, возникла необходимость загрузки одного или нескольких файлов во временную таблицу (форму) с последующей их вставкой в простую таблицу по отдельности в каждую строчку. Я пробовал делать так:

```
= :listReplace(list: $#nfv; action: $val; value: "value")
      val:insert(table: $#ntn; field: 't_file' = $filesdata; )
        filesdata: listCreate(item: $rowfile)
	  rowfile: rowCreate(field: "name" = $#value[name]; field: "filestring" = $#value[filestring];)
```
Но оно как-то не завелось.
Ещё был вариант добавить поле Данные, направить его на поле с файлами временной таблицы и кодом на изменение вставлять их в простую таблицу. Но при таком подходе похоже, что файлы затираются через час.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Пример для одно файла из поля в хедере. Код действия в кнопке в хедере:

```
= : insert(table: 'some_simple_table'; field: 'file_field' = $rowlist; log: true)

rowlist: listCreate(item: $row)
	row: rowCreate(field: "name" = #h_field_file[0][name]; field: "filestring" = $content)
		content: fileGetContent(file: #h_field_file[0][file])
```

Те для кода с listReplace:

```
= : istReplace(list: $#nfv; action: $val; value: "value")
      val: insert(table: $#ntn; field: 't_file' = $filesdata)
        filesdata: listCreate(item: $rowfile)
	  rowfile: rowCreate(field: "name" = $#value[name]; field: "filestring" = $content)
            content: fileGetContent(file: $#value[file])
``` 

Те `fileGetContent` получает содержимое файла — текст или бинарник, как это хранится в файловой системе.

**Comment by mihail-cornev** [comment]
@erpic-ru 
У меня похожая задача, но всё-же немного отличается:

1. Есть простая таблица с кнопкой Загрузить
2. По нажатию кнопки Загрузить запускаем linkToFileUpload
3. Загружаем несколько файлов и по нажатию кнопки ОК, кодом записываем каждый файл в отдельную строку во временной таблице
4. Добавляем вручную текстовое описание каждому файлу.
5. Потом по нажатию кнопки передаём файлы с описанием в простую таблицу.

Как вы решили задачу с несколькими файлами?

**Reply by ttm-alexeypolunin** [reply]
По идее listReplace решает этот вопрос. Те вы берете список ваших файлов из временной и обрабатываете их построчно.

**Reply by mihail-cornev** [reply]
у меня затык с загрузкой нескольких файлов в простую таблицу через linktofileupload. штудирую форум и доки.

такая конструкцию выдаёт ошибку:
```
a1= : linkToFileUpload(title: "Загрузить файлы"; limit: 50; code: $code; type: "*"; refresh: true)

```code:totum

a1=: listReplace(list: $#input; action: $val; value: "value")

      val: insert(table: 'files_'; field: 'file' = $filesdata)

        filesdata: listCreate(item: $rowfile)

	  	rowfile: rowCreate(field: "name" = $#value[name]; field: "filestring" = $content)

            		content: fileGetContent(file: $#value[file])
` ` `
```

ошибка:

```
Заполните параметр file.

Функция fileGetContent; Линия кода content.; Функция rowCreate; Линия кода rowfile.; Функция listCreate; Линия кода filesdata.; Insert field 'file'=$filesdata; Функция insert; Линия кода val.; Функция listReplace
```
